common/models/acl.js

```diff
@@ -506,12 +506,19 @@ module.exports = function(ACL) {
   };
 
   ACL.resolveRelatedModels = function() {
-    if (!this.roleModel) {
-      var reg = this.registry;
-      this.roleModel = reg.getModelByType(loopback.Role);
-      this.roleMappingModel = reg.getModelByType(loopback.RoleMapping);
-      this.userModel = reg.getModelByType(loopback.User);
-      this.applicationModel = reg.getModelByType(loopback.Application);
+    var self = this;
+    if (!self.roleModel) {
+      var reg = self.registry;
+      var builtInModels = {
+        roleModel: reg.getModel('Role'),
+        roleMappingModel: reg.getModel('RoleMapping'),
+        userModel: reg.getModel('User'),
+        applicationModel: reg.getModel('Application'),
+      };
+
+      for (var m in builtInModels) {
+        self[m] = reg.getModelByType(builtInModels[m]);
```

Comment:
@benkroeger I am still confused. Here is the code in registry that makes me think that those two calls should return the same result:

```js
Registry.prototype.getModelByType = function(modelType) {
  var type = typeof modelType;
  var accepted = ['function', 'string'];

  assert(accepted.indexOf(type) > -1,
    'The model type must be a constructor or model name');

  if (type === 'string') {
    modelType = this.getModel(modelType);
  }

  var models = this.modelBuilder.models;
  for (var m in models) {
    if (models[m].prototype instanceof modelType) {
      return models[m];
    }
  }
  return modelType;
};
```

As you can see, if you call `getModelByType('AccessToken')`, the function calls `getModel('AccessToken')` first before searching for descendant models.

I have also modified your patch and run the test suite to verify my assumptions, all of the tests passed.

```diff
diff --git a/common/models/acl.js b/common/models/acl.js
index 6d413a1..cdf931b 100644
--- a/common/models/acl.js
+++ b/common/models/acl.js
@@ -509,16 +509,10 @@ module.exports = function(ACL) {
     var self = this;
     if (!self.roleModel) {
       var reg = self.registry;
-      var builtInModels = {
-        roleModel: reg.getModel('Role'),
-        roleMappingModel: reg.getModel('RoleMapping'),
-        userModel: reg.getModel('User'),
-        applicationModel: reg.getModel('Application'),
-      };
-
-      for (var m in builtInModels) {
-        self[m] = reg.getModelByType(builtInModels[m]);
-      }
+      self.roleModel = reg.getModelByType('Role');
+      self.roleMappingModel = reg.getModelByType('RoleMapping');
+      self.userModel = reg.getModel('User');
+      self.applicationModel = reg.getModel('Application');
     }
   };
```

So my question is, what am I missing here?